pkg/bgp/speaker/pod_cidr.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of Cilium
+
+package speaker
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/cilium/cilium/pkg/cidr"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	metallbbgp "go.universe.tf/metallb/pkg/bgp"
+	metallbspr "go.universe.tf/metallb/pkg/speaker"
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	emptyAdverts = []*metallbbgp.Advertisement{}
+)
+
+func (s *MetalLBSpeaker) withDraw() error {
+	var wg sync.WaitGroup // waitgroup here since we don't care about errors
+	for _, session := range s.speaker.PeerSessions() {
+		wg.Add(1)
+		go func(sess metallbspr.Session) { // Need an outer closure to capture session.
+			defer wg.Done()
+			// providing an empty array or advertisements will
+			// provoke the BGP controller to withdrawal any currently
+			// advertised bgp routes.
+			err := sess.Set(emptyAdverts...)
+			if err != nil {
+				log.WithError(err).Error("Failed to withdraw BGP routes")
+			}
+		}(session)
+	}
+	wg.Wait()
+	return nil
+}
+
+func (s *MetalLBSpeaker) announcePodCIDRs(cidrs []string) error {
+	var eg errgroup.Group
+	for _, session := range s.speaker.PeerSessions() {
+		func(sess metallbspr.Session) { // Need an outer closure to capture session.
+			eg.Go(func() error {
+				err := s.announce(sess, cidrs)
+				if err == nil {
+					log.WithField(logfields.CIDR, cidrs).Debug("Announced Pod CIDRs")
```

Comment:
Would it be useful to log info about the session?

Reply:
Session is from the metallb api and has no informative data:

```
// Session gives access to the BGP session.
type Session interface {
	io.Closer
	Set(advs ...*bgp.Advertisement) error
}
```

Reply:
Actually kinda surprised to find that the session doesn't carry much useful info with it.

Reply:
@joamaki going to resolve this for now, but if I'm missing something feel free to un-resolve and ping me.